chain/network/src/peer_manager/tests/nonce.rs

```diff
@@ -74,3 +77,64 @@ async fn test_nonces() {
         }
     }
 }
+
+async fn wait_for_edge(actor_handler: &mut ActorHandler) -> Edge {
+    actor_handler
+        .events
+        .recv_until(|ev| match ev {
+            Event::PeerManager(peer_manager_actor::Event::EdgesVerified(ev)) => Some(ev[0].clone()),
+            _ => None,
+        })
+        .await
+}
+
+#[tokio::test]
+/// Create 2 peer managers, that connect to each other.
+/// Verify that the will refresh their nonce after some time.
+async fn test_nonce_refresh() {
+    init_test_logger();
+    let mut rng = make_rng(921853255);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::new(*EDGE_MIN_TIMESTAMP_NONCE + time::Duration::days(2));
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    // Start a PeerManager.
+    let pm = peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+
+    // Start another peer manager.
+    let mut pm2 = peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+
+    pm2.connect_to(&pm.peer_info()).await;
+
+    let edge = wait_for_edge(&mut pm2).await;
+    let start_time = clock.now_utc();
+    // First edge between them should have the nonce equal to the current time.
+    assert_eq!(Edge::nonce_to_utc(edge.nonce()).unwrap().unwrap(), start_time);
+
+    // Advance a clock by 1 hour.
+    clock.advance(Duration::HOUR);
+
+    loop {
+        let edge = wait_for_edge(&mut pm2).await;
+        if Edge::nonce_to_utc(edge.nonce()).unwrap().unwrap() == start_time {
+            tracing::debug!("Still seeing old edge..");
+            //
+            clock.advance(Duration::HOUR);
+        } else {
+            assert_eq!(Edge::nonce_to_utc(edge.nonce()).unwrap().unwrap(), clock.now_utc());
+            break;
+        }
```

Comment:
Instead of implementing this loop, you might just want to provide a more specific predicate to recv_until in the wait_for_edge call.

Reply:
I didn't want to overcomplicate the recv_until (plus here I actually check that the nonce didn't change in some strange way).